pennylane_cirq/qsim_device.py

```diff
@@ -0,0 +1,110 @@
+            to be >= 1. In analytic mode, shots indicates the number of entries
+            that are returned by ``device.sample``.
+        analytic (bool): indicates whether expectation values and variances should
+            be calculated analytically
+        qubits (List[cirq.Qubit]): A list of Cirq qubits that are used
+            as wires. The wire number corresponds to the index in the list.
+            By default, an array of ``cirq.LineQubit`` instances is created.
+    """
+    name = "QSim device for PennyLane"
+    short_name = "cirq.qsim"
+
+    def __init__(self, wires, shots=1000, analytic=True, qubits=None):
+        super().__init__(wires, shots, analytic, qubits)
+        self.circuit = None
+        self._simulator = qsimcirq.QSimSimulator()
+
+    def reset(self):
+        # pylint: disable=missing-function-docstring
+        super().reset()
+        self.circuit = qsimcirq.QSimCircuit(cirq_circuit=self.circuit)
+
+
+class QSimhDevice(SimulatorDevice):
+    r"""QSimh device for PennyLane.
+
+    Args:
+        wires (int, Iterable[Number, str]]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers (i.e., ``[-1, 0, 2]``)
+            or strings (``['ancilla', 'q1', 'q2']``). Default 1 if not specified.
+        shots (int): Number of circuit evaluations/random samples used
+            to estimate expectation values of observables. Shots need
+            to be >= 1. In analytic mode, shots indicates the number of entries
+            that are returned by ``device.sample``.
+        analytic (bool): indicates whether expectation values and variances should
+            be calculated analytically
+        qubits (List[cirq.Qubit]): A list of Cirq qubits that are used
+            as wires. The wire number corresponds to the index in the list.
+            By default, an array of ``cirq.LineQubit`` instances is created.
+        qsimh_options (dict): A dictionary with options for the QSimh simulator. See the `QSim
+            usage documentation <https://github.com/quantumlib/qsim/blob/master/docs/usage.md>`__
+            for further details.
+    """
+    name = "QSimh device for PennyLane"
+    short_name = "cirq.qsimh"
+
+    def __init__(self, wires, shots=1000, analytic=True, qubits=None, qsimh_options=None):
+        super().__init__(wires, shots, analytic, qubits)
+        if qsimh_options is None:
+            qsimh_options = {"k": [0], "w": 0, "p": 0, "r": 1}
+        self.circuit = None
+        self.qsimh_options = qsimh_options
+        self._simulator = qsimcirq.QSimhSimulator(qsimh_options)
+
+    def apply(self, operations, **kwargs):
+        CirqDevice.apply(self, operations, **kwargs)
+
+        # TODO: remove the need for this hack by keeping better track of unused wires
+        # We apply identity gates to all wires, otherwise Cirq would ignore
+        # wires that are not acted upon
```

Comment:
Is this todo copied from the existing Cirq device? Looks familiar

Reply:
Yes, it was written differently before, but I noticed the `TODO:` in the existing Cirq device, so I thought it might be good to duplicate it here so we don't forget about this one as well (unfortunately this  part needed to be duplicated).

Reply:
@josh146 Would you recommend just keeping the `TODO`s here and simply ignore the pylint complaint for this PR?

Reply:
Yep, fine to simply ignore pylint here!